yt_dlp/extractor/yandexvideo.py

```diff
@@ -260,11 +260,12 @@ def _real_extract(self, url):
         data_json = self._search_json(
             r'("data"\s*:|data\s*=)', webpage, 'metadata', video_id, contains_pattern=r'{["\']_*serverState_*video.+}')
         serverstate = self._search_regex(r'(_+serverState_+video-site_[^_]+_+)',
-                                         webpage, 'server state').replace('State', 'Settings')
+                                         webpage, 'server state')
         uploader = self._search_regex(r'(<a\s*class=["\']card-channel-link[^"\']+["\'][^>]+>)',
                                       webpage, 'uploader', default='<a>')
         uploader_name = extract_attributes(uploader).get('aria-label')
-        video_json = try_get(data_json, lambda x: x[serverstate]['exportData']['video'], dict)
+        item_id = try_get(data_json, lambda x: x[serverstate]['videoViewer']['openedItemId'], str)
+        video_json = try_get(data_json, lambda x: x[serverstate]['videoViewer']['items'][item_id], dict)
```

Comment:
Let's use new/better helpers for edited code

```suggestion
        item_id = traverse_obj(data_json, (serverstate, 'videoViewer', 'openedItemId', {str}))
        video_json = traverse_obj(data_json, (serverstate, 'videoViewer', 'items', item_id, {dict}))
```